### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by cp-pratik-k**
### Purpose
 - Clean up

### Summary of Changes
 - Clean up

### Test steps
- Run flutter test

### Conformity
- [x] Followed [git guidelines](https://github.com/canopas/flutter-developer-roadmap/blob/main/GitGuideline.md) for creating commit messages and Pull Request guidelines.
- [x] Self-approved the PR - reviewed the PR as a reviewer and gave it self-approval if everything is ok. If not, made the required changes.
- [x] Ensured that the PR satisfies all specified requirements in the ticket, including bug fixes and new features.
- [x] Provided test steps, including steps to reproduce the issue or test the new functionality, ensuring other team members can verify the changes.
- [x] Added/Updated proper code comments to make it easy-to-understand for other developers.
- [x] Reused code (if the same code was written twice, made it common and reused it at both places).
- [x] Removed unused or commented code if not required.
- [x] Ensured proper Dart naming conventions were used for variables, classes, and methods.
- [x] Localized user-facing strings.
- [x] Included screenshots/videos of behavior changes: Provided visual evidence of any changes to UI or behavior for easier review and understanding in the PR description.
- [x] Implemented proper error handling: Ensured that the code anticipated and handled potential errors and edge cases gracefully.
- [x] Avoided introducing technical debt: If the PR introduces technical debt, created and linked appropriate tickets for future resolution.
- [x] Included relevant unit tests: Wrote unit tests that focused on testing behavior and functionality, rather than merely covering lines of code.
- [x] Ensured code was performant and scalable: Verified that the changes did not introduce performance issues or bottlenecks and could scale as needed.
- [x] Ensured comments were up-to-date and relevant to the code to describe complex logic and to add understanding for other developers.
- [x] Marked the PR as ready before submitting it for review.

### CODE REVIEW

File: lib/data/di/app_module.dart [hunk 1]
```diff
@@ -8,17 +8,22 @@ import 'package:injectable/injectable.dart';
...
+  ImagePicker get imagePicker => ImagePicker();
+
   FirebaseStorage get firebaseStorage => FirebaseStorage.instance;
+
   FirebaseFirestore get firebaseFireStore => FirebaseFirestore.instance;
+
   FirebaseAuth get firebaseAuth => FirebaseAuth.instance;
+
   DeviceInfoPlugin get deviceInfoPlugin => DeviceInfoPlugin();
-  http.Client get httpClient => http.Client();
```
Comment (cp-sneha-s) [inline lib/data/di/app_module.dart:27]: Let's add singleton of `FirestoreCrashlytics`

File: lib/data/services/leave_service.dart [hunk 2]
```diff
@@ -25,13 +24,6 @@ class LeaveService {
             toFirestore: (Leave leave, _) => leave.toFireStore(leave));
   }
```
Comment (cp-sneha-s) [inline lib/data/services/leave_service.dart:25]: ` fireStore .collection(FireStoreConst.spaces) .doc(_userManager.currentSpaceId!)`
 Let's remove docId from the service class `_userManager.currentSpaceId!`
Change it here and everywhere in service class.

Comment (cp-pratik-k) [reply lib/data/services/leave_service.dart:25]: we can do it that after all branch marged. otherwise its should show so many conflict in our user branch.

**Review by cp-sneha-s — APPROVED** [review]
